File: 27_remove_element/remove_element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        nums[:] = [nums[i] for i in range(0, len(nums)) if nums[i] != val]
        return len(nums)



if __name__ == '__main__':
    sol = Solution()
    val = 1
    nums = [1, 1, 2, 4, 5, 1, 2]
    result = sol.removeElement(nums, val)
    print nums, "&", result

File: 27_remove_element/removeElement.cpp
#include <iostream>
#include <vector>

using namespace std;


class Solution {
	public:
    	int removeElement(vector<int>& nums, int val) {
            int begin = -1, forward = -1;
            while(++forward < nums.size()) {
                if(nums[forward] != val)
                    nums[++begin] = nums[forward];
            }
            nums.erase(nums.begin() + begin + 1, nums.end());
            return nums.size();
    	}
};

int main() {
    return 0;
}
